Automattic/jetpack
# Search: Add connection screen for search dashboard

Fixes #23154 

#### Changes proposed in this Pull Request:
Added connection page for the search dashboard when the site is not connected. ATM when site is connected but has no search product, the dashboard is showing like the following, which is not ideal but works to some extent as it has the upgrade button. We'll add another screen for user to purchase Search product in the following beta project.

The PR boots up `My Jetpack` too, which handles the disconnection process and  more.


Note:

- The pricing is hard coded for now.
- Not all designs applied - following up PRs will be proposed to apply designs, but not sure this would fall in our Alpha plugin scope.
- Known issue: the returning URL after purchase might not be right (sometimes ends up with /wp-admin/admin.php?page=jetpack#/recommendations), which will be looked into in a follow up PR.
- D77619 provides a hack to walk around the issue where sites are not recognized as Jetpack sites. And we are currently discussing with the crew to see the best solution.


#### Jetpack product discussion
pcNPJE-Ch-p2

#### Does this pull request change what data or activity we track or use?
no

#### Testing instructions:
- Apply patch `D77619`
- Sandbox `public-api.wordpress.com`
- Go to a site with NO Jetpack product subscription ( you could remove from SA )
- Sandbox Jetpack
    - In your `wp-config.php`, you’ll tell Jetpack to talk to your sandbox using `define( 'JETPACK__SANDBOX_DOMAIN', '$SANDBOX_HOSTNAME' );`
    - Enable sandbox global access `sudo global-http enable`
- Disconnect and deactivate Jetpack
- Ensure you could click `Get Jetpack Search` button to connect site and purchase a subscription
- Disconnect Jetpack thru `My Jetpack`
- Click login, ensure you can connect the site
- Ensure Search dashboard works
- Ensure other search feature works

![image](https://user-images.githubusercontent.com/1425433/159369705-274d0a52-b87a-411b-a0e9-95f400694212.png)

![image](https://user-images.githubusercontent.com/1425433/160318999-f516f7a7-2f61-4b35-91bc-8f7518aea9c2.png)

![image](https://user-images.githubusercontent.com/1425433/159604072-e01a75be-ef38-4d3b-9e56-a005c0ebd35b.png)
